### PR TITLE
Appboy to Braze rename, misc refactoring

### PR DESCRIPTION
**Pull request by pramttl**
- AppboyClient -> BrazeClient
- Allow specifying base_url while instantiating class
- pytestify tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,16 +1,6 @@
-<p align="center">
-  <a href="https://hellofresh.com">
-    <img width="120" src="https://www.hellofresh.de/images/hellofresh/press/HelloFresh_Logo.png">
-  </a>
-</p>
-
-# appboy-python-client
+# braze-client
 A Python client for the Appboy REST API
```
Comment (AA33) [inline README.md]: Braze REST API?

File: braze/__init__.py [hunk 2]
```diff
@@ -0,0 +1 @@
+from .client import BrazeClient
```
Comment (AA33) [inline braze/__init__.py]: Let's keep `__init__` empty.

File: README.md [hunk 3]
```diff
@@ -24,9 +14,8 @@ $ python setup.py install
 ### How to use
 
 ```python
-from appboy.client import AppboyClient
-
-client = AppboyClient(api_key='YOUR_API_KEY')
+from braze import BrazeClient
```
Comment (AA33) [inline README.md]: Update this once init is updated. See below.

File: braze/client.py [hunk 4]
```diff
@@ -4,12 +4,12 @@
...
 
-class AppboyClient(object):
+class BrazeClient(object):
     """
     Client for Appboy public API. Support user_track.
     usage:
-     from appboy.client import AppboyClient
-     client = AppboyClient(app_group_id='Place your app_group_id here')
+     from braze.client import BrazeClient
+     client = BrazeClient(app_group_id='Place your app_group_id here')
```
Comment (AA33) [inline braze/client.py]: API Key, instead of `app_group_id`.

File: examples.py [hunk 5]
```diff
@@ -1,6 +1,6 @@
-from appboy.client import AppboyClient
+from braze import BrazeClient
 
-client = AppboyClient(api_key='YOUR_API_KEY')
+client = BrazeClient(app_group_id='YOUR_GROUP_ID')
```
Comment (AA33) [inline examples.py]: We're using `api_key` now.

File: tests/test_client.py [hunk 6]
```diff
@@ -32,7 +32,7 @@ def json():
 
 class TestAppboyClient(unittest.TestCase):
     def setUp(self):
-        self.client = AppboyClient(api_key='API_KEY')
+        self.client = BrazeClient(app_group_id='APP_GROUP_ID')
```
Comment (AA33) [inline tests/test_client.py]: Again we're using `api_key` now.

Comment (pramttl) [reply tests/test_client.py]: oh, did the rename before rebasing over the latest master and then rebased. Will update.

File: braze/client.py [hunk 7]
```diff
@@ -28,15 +28,15 @@ class AppboyClient(object):
...
+    DEFAULT_API_URL = 'https://api.appboy.com'
     USER_TRACK_ENDPOINT = '/users/track'
     USER_DELETE_ENDPOINT = '/users/delete'
 
     REQUEST_POST = 'post'
 
-    def __init__(self, api_key):
+    def __init__(self, api_key, api_url=None):
         self.api_key = api_key
+        self.api_url = api_url or self.DEFAULT_API_URL
```
Comment (pramttl) [inline braze/client.py]: Braze may give different customers (such as enterprise customers) a different base endpoint. This change allows the api_url to be specified when instantiating a client instance. If missing it defaults to `https://api.appboy.com`.

File: tox.ini [hunk 8]
```diff
@@ -16,17 +16,17 @@ deps =
...
-commands = {[testenv:unit]commands}
+deps = {[testenv]deps}
+commands = {[testenv]commands}
 
 # Codestyle
 [testenv:style]
 deps = flake8
-commands = flake8 --max-line-length=120 appboy
\ No newline at end of file
+commands = flake8 --max-line-length=120 braze
```
[the remark this answers is not in the document]
Comment (pramttl) [reply tox.ini]: yes, I do hope to add black and that should make the style consistent but in a separate PR to not clutter the diff on this one.

**Pull request merged**